src/main/java/io/cdap/plugin/gcp/dataplex/common/util/DataplexUtil.java

```diff
@@ -0,0 +1,414 @@
+
+        // Record names have to be unique as Avro doesn't allow to redefine them.
+        // We can make them unique by prepending the previous records names to their name.
+        String recordName = "";
+        if (recordPrefix != null) {
+          recordName = recordPrefix + '.';
+        }
+        recordName = recordName + field.getName();
+
+        for (com.google.cloud.dataplex.v1.Schema.SchemaField f : fields) {
+          Schema.Field schemaField = getSchemaField(f, collector, recordName);
+          // if schema field is null, that means that there was a validation error. We will still continue in order to
+          // collect more errors
+          if (schemaField == null) {
+            continue;
+          }
+          schemaFields.add(schemaField);
+        }
+        // do not return schema for the struct field if none of the nested fields are of supported types
+        if (!schemaFields.isEmpty()) {
+          return Schema.recordOf(recordName, schemaFields);
+        } else {
+          return null;
+        }
+      default:
+        String error =
+          String.format("Entity column '%s' is of unsupported type '%s'.", field.getName(), standardType.name());
+        String action = String.format("Supported column types are: %s.",
+          BigQueryUtil.BQ_TYPE_MAP.keySet().stream().map(t -> t.getStandardType().name())
+            .collect(Collectors.joining(", ")));
+        if (collector != null) {
+          collector.addFailure(error, action);
+        } else {
+          throw new RuntimeException(error + action);
+        }
+        return null;
+    }
+  }
+
+  /**
+   * Converts Entity schema for partitioned field into a corresponding CDAP Schema.Field.
+   *
+   * @param field        Partitioned field to be converted.
+   * @param collector    Failure collector to collect failure messages for the client.
+   * @param recordPrefix String to prepend to recordNames to make them unique
+   * @return A CDAP schema field
+   */
+  @Nullable
+  private static Schema.Field getSchemaField(com.google.cloud.dataplex.v1.Schema.PartitionField field,
+                                             @Nullable FailureCollector collector,
+                                             @Nullable String recordPrefix) {
+    Schema schema = convertFieldTypeForPartitionedFields(field, collector, recordPrefix);
+    if (schema == null) {
+      return null;
+    }
+    return Schema.Field.of(field.getName(), Schema.nullableOf(schema));
+  }
+
+  /**
+   * Converts Entity Partitioned field type into a CDAP field type.
+   *
+   * @param field        Partitioned field to be converted.
+   * @param collector    Failure collector to collect failure messages for the client.
+   * @param recordPrefix String to add before a record name to ensure unique names.
+   * @return A CDAP field schema
+   */
+  @Nullable
+  private static Schema convertFieldTypeForPartitionedFields(com.google.cloud.dataplex.v1.Schema.PartitionField field,
```

Comment:
there is similar method `convertFieldTypeForPartitionedFields`, where difference is it is partitioned field or not, can this be refactored to some common methods?

Reply:
done